uploadNeuroDB/NeuroDB/MRI.pm

```diff
@@ -548,18 +551,45 @@ sub identify_scan_db {
         $ScannerID=$results[0];
     }
 
-    
-    # get the list of protocols for a site their scanner and subproject
-    $query = "SELECT Scan_type, ScannerID, Center_name, TR_range, TE_range, TI_range, slice_thickness_range, xspace_range, yspace_range, zspace_range,
-              xstep_range, ystep_range, zstep_range, time_range, series_description_regex
-              FROM mri_protocol
-              WHERE
-             (Center_name='$psc' AND ScannerID='$ScannerID')
-              OR ((Center_name='ZZZZ' OR Center_name='AAAA') AND ScannerID='0')
-              ORDER BY Center_name ASC, ScannerID DESC";
+    #===========================================================#
+    # Get the list of lines in the mri_protocol table that      #
+    # apply to the given scan based on the center name, project # 
+    # ID, subproject ID and visit label                         #
+    #===========================================================#
+    ($query = <<QUERY) =~ s/\n/ /gm;
+    SELECT mp.Scan_type, mp.ScannerID, mp.Center_name, 
+           mp.TR_range, mp.TE_range, mp.TI_range, mp.slice_thickness_range,
+           mp.xspace_range, mp.yspace_range, mp.zspace_range,
+           mp.xstep_range, mp.ystep_range, mp.zstep_range, 
+           mp.time_range, mp.series_description_regex
+    FROM mri_protocol mp
+    JOIN mri_protocol_group_rel mpgr ON (mp.ID=mpgr.MriProtocolID)
+    JOIN mri_protocol_group_target mpgt USING(MriProtocolGroupID)
+    WHERE
+        (Center_name = ? AND ScannerID = ?)
+    OR  ((Center_name='ZZZZ' OR Center_name='AAAA') AND ScannerID='0')
+QUERY
 
+    $query .= defined $projectID
+        ? ' AND (mpgt.ProjectID IS NULL OR mpgt.ProjectID = ?)'
+        : ' AND mpgt.ProjectID IS NULL';
+    $query .= defined $subprojectID
+        ? ' AND (mpgt.SubprojectID IS NULL OR mpgt.SubprojectID = ?)'
+        : ' AND mpgt.SubprojectID IS NULL';
+    $query .= defined $visitLabel
+        ? ' AND (mpgt.Visit_label IS NULL OR mpgt.Visit_label = ?)'
+        : ' AND mpgt.Visit_label IS NULL';
```

Comment:
Let's say you have 2 t1 protocols (one with the default protocol and one specific to a project):
- t1 default protocol: TR=2000-2400
- t1 project X protocol: TR=2100
What happens if for project X you try to insert a t1 with TR=2200. Theoretically, you don't want it to be inserted but wouldn't it get inserted anyway because of the t1 default protocol that would allow for its insertion?

Reply:
haha. OK. Will wait for the next commit then. Thanks!